xformers/factory/block_factory.py

```diff
@@ -130,7 +130,7 @@ def __init__(self, config: xFormerEncoderConfig):
     def from_config(cls, config: xFormerEncoderConfig, **kwargs):
         return cls(config, **kwargs)
 
-    def forward(self, x: torch.Tensor):
+    def forward(self, x: torch.Tensor, *args, **kwargs):
```

Comment:
```suggestion
    def forward(self, x: torch.Tensor, input_mask=None):
```

Let's do this then to ensure other weird sh*t doesn't go inside unnoticed.

Reply:
```suggestion
    def forward(
        self,
        x: torch.Tensor,
        att_mask: Optional[Union[torch.Tensor, AttentionMask]] = None,
        input_mask: Optional[torch.Tensor] = None,
    ):
```
In this case I suggest we restore the full signature. Btw, I don't think we use or need `xFormerDecoderBlock`, but I'll note here that the signature of its forward function is incompatible as well:
```python
    def forward(
        self,
        target: torch.Tensor,
        memory: torch.Tensor,
        encoder_att_mask: Optional[Union[torch.Tensor, AttentionMask]] = None,
        decoder_att_mask: Optional[Union[torch.Tensor, AttentionMask]] = None,
        input_mask: Optional[torch.Tensor] = None,
    ):
```

Reply:
Let's restore the full signature yes